backends/ebpf/tests/test.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+# Copyright 2022-present Orange
+# Copyright 2022-present Open Networking Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function print_help() {
+  # Display Help
+  echo "Run PTF tests for PSA-eBPF."
+  echo "The script will run all combinations of flags if no options are provided."
+  echo
+  echo "Syntax: ./test.sh [OPTIONS] [TEST_CASE]"
+  echo
+  echo "OPTIONS:"
+  echo "--bpf-hook       A BPF hook that should be used as a main attach point <tc|xdp>"
+  echo "--xdp2tc         A mode to pass metadata from XDP to TC programs <meta|head|cpumap>."
+  echo "--trace          Build P4 programs with tracing logs (disabled by default) <on|off>."
+  echo "--help           Print this message."
+  echo
+}
+
+function exit_on_error() {
+      exit_code=$?
+      if [ $exit_code -ne 0 ]; then
+          exit $exit_code
+      fi
+}
+
+function cleanup() {
+      echo "Cleaning...."
+      for intf in "${INTERFACES[@]}" ; do
+        ip link del "s1-$intf"
+      done
+      ip netns exec switch ip link del psa_recirc
+      ip netns exec switch ip link del psa_cpu
+      ip netns pids switch | (xargs kill 2>/dev/null)
+      ip netns del switch
+      # remove all pinned eBPF objects
+      rm -rf /sys/fs/bpf/*
+      echo "Cleaning finished"
+}
+
+if [ "x$1" = "x--help" ]; then
+  print_help
+  exit 0
+fi
+
+for i in "$@"; do
+  case $i in
+    --bpf-hook=*)
+      BPF_HOOK="${i#*=}"
+      shift # past argument=value
+      ;;
+    --xdp2tc=*)
+      XDP2TC_ARG="${i#*=}"
+      shift # past argument=value
+      ;;
+    --trace=*)
+      TRACE_LOGS_ARGS="${i#*=}"
+      shift # past argument=value
+      ;;
+    *)
+      # unknown option
+      ;;
+  esac
+done
+
+cleanup
+trap cleanup EXIT
+
+# Remove any temporary files from previous run. It might be useful to
+# preserve these files after test run for inspection
```

Comment:
yes, the p4c test scripts have an option -b to preserve the temporary files

Reply:
Yes, we preserve output files for failing tests.